### PR TITLE
Go to space landing page when clicking on a selected space

### PR DESCRIPTION
**Pull request by robintown**
Context for design: In the spaces feedback room, [Matthew raised the point](https://matrix.to/#/!LPqOcoJQuGkweiTedq:riot.ovh/$Sna3Ry9-0DoAt2qGfYxM-T0M56kwnwXxCNJU9pMgQYA?via=matrix.org&via=fairydust.space&via=envs.net) that the explore rooms page still feels a bit too hidden and difficult to get to. My proposed solution, which this implements, was to make clicking on a space that's already selected pull up the explore rooms page.

Fixes https://github.com/vector-im/element-web/issues/20296.

Type: enhancement

---
Here's what your changelog entry will look like:

## ✨ Features
 * Go to space landing page when clicking on a selected space ([\#6442](https://github.com/matrix-org/matrix-react-sdk/pull/6442)). Fixes vector-im/element-web#20296.

### CODE REVIEW

**Review by gsouquet — APPROVED** [review]
That makes sense code-wise. 
Waiting on product feedback whether we want to enable that behaviour

**Comment by niquewoodhouse** [comment]
I know its listed as needing product to review it, but just to say I find this a little confusing when I have `People` or `Favourites` selected, it just goes to my Home

**Comment by robintown** [comment]
I may end up removing the 'metaspace -> home' behavior entirely given the work that Delight is now doing in this area

**Review by kittykat — CHANGES_REQUESTED** [review]
From a product POV: Danielle, Nique and I discussed this and we feel it should apply only to real spaces (not metaspaces). Otherwise, looks good.

**Comment by kittykat** [comment]
Found this issue when testing: https://github.com/vector-im/element-web/issues/22156

I experience it on develop as well, but I can't test the build properly until it's resolved :(

**Review by kittykat — APPROVED** [review]
LGTM

**Comment by robintown** [comment]
Requesting re-review from the team since it's been a while

File: src/components/views/spaces/SpaceTreeLevel.tsx [hunk 1]
```diff
@@ -103,6 +105,10 @@ export const SpaceButton: React.FC<IButtonProps> = ({
         />;
     }
 
+    const onClick = props.onClick ?? (selected && space
+        ? () => defaultDispatcher.dispatch({ action: Action.ViewRoom, room_id: space.roomId })
+        : () => SpaceStore.instance.setActiveSpace(spaceKey ?? space.roomId));
```
Comment (t3chguy) [inline src/components/views/spaces/SpaceTreeLevel.tsx:110]: this is a little difficult to read, can we break it up?

**Review by t3chguy — APPROVED** [review]
LGTM otherwise

**Pull request merged**